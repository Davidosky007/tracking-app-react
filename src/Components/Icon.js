import PropTypes from 'prop-types';
import { Link, useLocation } from 'react-router-dom';

const Icon = (props) => {
  const {
    path, title, clickHandler, icon,
  } = props;

  const location = useLocation();
  const route = location.pathname.split('/')[1];

  const isActive = () => {
    if (`/${route}` === path) return 'blue';
    return 'dark-gray';
  };
  const active = isActive();
  return (
    <Link className={`icon background-${active} color-white`} to={path} onClick={clickHandler}>
      <div>
        <div className={`${icon} big`} />
        <div className="small m-top-10">{title}</div>
      </div>
    </Link>
  );
};

Icon.propTypes = {
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  clickHandler: PropTypes.func,
  icon: PropTypes.string.isRequired,
};

Icon.defaultProps = {
  clickHandler: () => {},
};

export default Icon;
