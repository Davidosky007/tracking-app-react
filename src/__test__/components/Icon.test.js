import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Icon from '../../Components/Icon';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'example.com/test',
  }),
}));

describe('Icon', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Icon
          path=""
          title=""
          clickHandler={() => {}}
          icon=""
        />
      </BrowserRouter>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
