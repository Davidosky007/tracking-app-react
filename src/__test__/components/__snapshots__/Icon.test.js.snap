// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<a
  className="icon background-dark-gray color-white"
  href="/"
  onClick={[Function]}
>
  <div>
    <div
      className=" big"
    />
    <div
      className="small m-top-10"
    >
      
    </div>
  </div>
</a>
`;
